contracts/parameterized/Parameterized.sol

```diff
@@ -0,0 +1,134 @@
+///         - UUID: a unique identifier         
+
+contract Parameterized is ITransferDefinition {
+  using LibChannelCrypto for bytes32;
+  using SafeMath for uint256;
+
+  // Rate = deltaAmount/deltaTime
+  struct Rate {
+    uint256 deltaAmount;
+    uint256 deltaTime;
+  }
+
+  // TransferState contains the payment data
+  struct TransferState {
+    address receiver;
+    uint256 start;
+    uint256 expiration;
+    bytes32 UUID;
+    Rate rate;
+  }
+
+  // Encapsulates the UUID and payment amount to make signing easier - see below TransferResolver
+  struct ResolverData {
+    bytes32 UUID;
+    uint256 paymentAmountTaken;
+  }
+
+  // TransferResolver contains the UUID of the payment we're signing to prevent replay attacks,
+  // the payment amount taken, and a signature of this data
+  struct TransferResolver {
+    ResolverData data;
+    bytes payeeSignature;
+  }
+
+  /* solhint-disable */
+  string StateEncoding = "tuple(address receiver, uint256 start, uint256 expiration, bytes32 UUID, tuple(uint256 deltaAmount, uint256 deltaTime) rate)";
+  string ResolverEncoding = "tuple(tuple(bytes32 UUID, uint256 paymentAmountTaken) data, bytes payeeSignature)";
+  string Name = "Parameterized";
+  /* solhint-enable */
+
+  function getRegistryInformation() external override view returns (RegisteredTransfer memory) {
+    RegisteredTransfer memory info = RegisteredTransfer({
+      name: Name,
+      stateEncoding: StateEncoding,
+      resolverEncoding: ResolverEncoding,
+      definition: address(this)
+    });
+    return info;
+  }
+
+  /// @notice Creates a parameterized payment from the payer (creator) to the payee (resolver)
+  /// @param encodedBalance balance the creator/payer is putting into the transfer
+  /// @param encodedState encoded transfer parameters
+  /// @dev encoded parameter data contains the receiver address, the expiration, the UUID, and the rate
+  function create(
+    bytes calldata encodedBalance,
+    bytes calldata encodedState
+  ) external override view returns (bool) {
+    // Unpack balance & state
+    Balance memory balance = abi.decode(encodedBalance, (Balance));
+    TransferState memory state = abi.decode(encodedState, (TransferState));
+    Rate memory rate = state.rate;
+
+    // Receiver should not have an initial balance within the transfer
+    require(balance.amount[1] == 0, "Cannot create parameterized payment with nonzero recipient init balance");
+
+    // Parameterized payment checks
+    require(rate.deltaAmount >= 1, "Per-unit amount must be at least 1 wei");
+    require(rate.deltaTime >= 1, "Per-unit time must be at least 1 second");
+    require(state.expiration > block.timestamp + 3 days, "Expiration must be at least 3 days in the future.");
+
+    // Sanity checks
+    require(state.receiver != address(0x0), "Receiver address cannot be the zero address!");
+    require(state.UUID != bytes32(0), "UUID cannot be null.");
+
+    return true;
```

Comment:
Should there be any sensible checks about the `state.start`? (This function is the only time any of the data in the transfer state will be validated, unless external validation is provided to the engine -- we can talk about how you would do that, but I would recommend putting validation in here as well)

Reply:
@LayneHaber I thought about this, but I'd like to preserve the ability to "backdate" payments arbitrarily far back in the past. An application that creates a Parameterized transfer should consider the start, stop, and rate together when determining what parameters to supply.